core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -0,0 +1,717 @@
+   * @param tablePath
+   */
+  public static void removeTempFolder(Map<String, FolderDetails> locationMap, String tmpFolder,
+      String tablePath) {
+    if (locationMap == null) {
+      return;
+    }
+    for (Map.Entry<String, SegmentFileStore.FolderDetails> entry : locationMap.entrySet()) {
+      String location = entry.getKey();
+      if (entry.getValue().isRelative()) {
+        location = tablePath + CarbonCommonConstants.FILE_SEPARATOR + location;
+      }
+      CarbonFile oldFolder =
+          FileFactory.getCarbonFile(location + CarbonCommonConstants.FILE_SEPARATOR + tmpFolder);
+      if (oldFolder.exists()) {
+        FileFactory.deleteAllCarbonFilesOfDir(oldFolder);
+      }
+    }
+  }
+
+  /**
+   * Returns content of segment
+   * @return
+   */
+  public Map<String, FolderDetails> getLocationMap() {
+    if (segmentFile == null) {
+      return new HashMap<>();
+    }
+    return segmentFile.getLocationMap();
+  }
+
+  /**
+   * Returs the current partition specs of this segment
+   * @return
+   */
+  public List<PartitionSpec> getPartitionSpecs() {
+    List<PartitionSpec> partitionSpecs = new ArrayList<>();
+    if (segmentFile != null) {
+      for (Map.Entry<String, FolderDetails> entry : segmentFile.getLocationMap().entrySet()) {
+        String location = entry.getKey();
+        if (entry.getValue().isRelative) {
+          location = tablePath + CarbonCommonConstants.FILE_SEPARATOR + location;
+        }
+        if (entry.getValue().getStatus().equals(SegmentStatus.SUCCESS.getMessage())) {
+          partitionSpecs.add(new PartitionSpec(entry.getValue().partitions, location));
+        }
+      }
+    }
+    return partitionSpecs;
+  }
+
+  /**
+   * It contains the segment information like location, partitions and related index files
+   */
+  public static class SegmentFile implements Serializable {
+
+    private static final long serialVersionUID = 3582245668420401089L;
+
+    private Map<String, FolderDetails> locationMap;
+
+    public SegmentFile merge(SegmentFile mapper) {
+      if (this == mapper) {
+        return this;
+      }
+      if (locationMap != null && mapper.locationMap != null) {
+        for (Map.Entry<String, FolderDetails> entry : mapper.locationMap.entrySet()) {
+          FolderDetails folderDetails = locationMap.get(entry.getKey());
+          if (folderDetails != null) {
+            folderDetails.merge(entry.getValue());
+          } else {
+            locationMap.put(entry.getKey(), entry.getValue());
+          }
+        }
+      }
+      if (locationMap == null) {
+        locationMap = mapper.locationMap;
+      }
+      return this;
+    }
+
+    public Map<String, FolderDetails> getLocationMap() {
+      return locationMap;
+    }
+
+    public void setLocationMap(Map<String, FolderDetails> locationMap) {
+      this.locationMap = locationMap;
+    }
+  }
+
+  /**
+   * Represents one partition folder
+   */
+  public static class FolderDetails implements Serializable {
```

Comment:
rename FolderDetails to PartitionFolderDetails

Reply:
I feel better we can keep as FolderDetails only because this concept can be extended normal table also.